back/engines/commercial/analysis/app/lib/analysis/LLM/base.rb

```diff
@@ -11,6 +11,11 @@ def name
         self.class.name.demodulize.underscore
       end
 
+      # Should return a family key as defined in LLMSelector::FAMILIES
+      def family
+        raise NotImplementedError
+      end
```

Comment:
A method that returns the family could be convenient.
```suggestion
      def family_key
        raise NotImplementedError
      end
      
      def family
        LLMSelector::FAMILIES.find { |family| family.key == family_key }
      end
```